packages/components/src/icon-button/index.js

```diff
@@ -45,7 +45,7 @@ class IconButton extends Component {
 
 		let element = (
 			<Button aria-label={ label } { ...additionalProps } className={ classes }>
-				{ isString( icon ) ? <Dashicon icon={ icon } ariaPressed={ ariaPressed } /> : icon }
+				{ isString( icon ) ? <Dashicon icon={ icon } ariaPressed={ ariaPressed } { ...additionalProps } /> : icon }
```

Comment:
Can it cause any issues when the same props get passed down to `Button` and `Dashicon`?

This one should be tested:
https://github.com/WordPress/gutenberg/blob/master/packages/components/src/form-token-field/token.js#L72

Reply:
Thank you @gziolo , that's a good concern.

Should we pass a specific `iconProps`?
I think it looks more declarative if we do this:
```jsx
<Button aria-label={ label } { ...additionalProps } className={ classes }>
	{ isString( icon ) ? <Dashicon icon={ icon } ariaPressed={ ariaPressed } { ...iconProps } /> : icon }
	{ children }
</Button>
```
From the caller:
```jsx
<ToolbarButton
	label={ __( 'Add block' ) }
	icon="plus-alt"
	onClick={ onInsertClick }
	iconProps={ { style: styles.addBlockButton, color: styles.addBlockButton.color } }
/>
```

What other solution do you think might be good?